### ISSUE TITLE
gcc 5.4 build error

### ISSUE THREAD

**Issue by abraxa** [mxe/mxe#1920]
I followed https://sigrok.org/gitweb/?p=sigrok-util.git;a=blob;f=cross-compile/mingw/README, which basically tells to fetch MXE from git, apply a one-liner patch for libzip and then build using
"make MXE_TARGETS=i686-w64-mingw32.static.posix gcc glib libzip libusb1 libftdi1 glibmm qt5 boost check". Doing so fails when building gcc 5.4:

```
/bin/sh /home/abraxa/repositories/sigrok/windows_build/mxe-git/tmp-gcc-i686-w64-mingw32.static.posix/gcc-5.4.0/gcc/../mkinstalldirs /home/abraxa/repositories/sigrok/windows_build/mxe-git/usr/lib/gcc/i686-w64-mingw32.static.posix/5.4.0/include
rm -rf /home/abraxa/repositories/sigrok/windows_build/mxe-git/usr/lib/gcc/i686-w64-mingw32.static.posix/5.4.0/include-fixed
mkdir /home/abraxa/repositories/sigrok/windows_build/mxe-git/usr/lib/gcc/i686-w64-mingw32.static.posix/5.4.0/include-fixed
chmod a+rx /home/abraxa/repositories/sigrok/windows_build/mxe-git/usr/lib/gcc/i686-w64-mingw32.static.posix/5.4.0/include-fixed
(cd `${PWDCMD-pwd}`/include ; \
 tar -cf - .; exit 0) | (cd /home/abraxa/repositories/sigrok/windows_build/mxe-git/usr/lib/gcc/i686-w64-mingw32.static.posix/5.4.0/include; tar xpf - )
tar: ./mmintrin.h: Cannot open: Permission denied
tar: ./fmaintrin.h: Cannot open: Permission denied
tar: ./fxsrintrin.h: Cannot open: Permission denied
tar: ./avxintrin.h: Cannot open: Permission denied
tar: ./xsaveoptintrin.h: Cannot open: Permission denied
tar: ./rtmintrin.h: Cannot open: Permission denied
tar: ./bmiintrin.h: Cannot open: Permission denied
tar: ./xsavecintrin.h: Cannot open: Permission denied
tar: ./bmi2intrin.h: Cannot open: Permission denied
tar: ./stdfix.h: Cannot open: Permission denied
tar: ./avx512fintrin.h: Cannot open: Permission denied
tar: ./varargs.h: Cannot open: Permission denied
tar: ./stddef.h: Cannot open: Permission denied
tar: ./lzcntintrin.h: Cannot open: Permission denied
```

Full build log is attached. Could it be that `${PWDCMD-pwd}` evaluates to an empty string, making the script try to untar the archive into /include?

[gcc_i686-w64-mingw32.static.posix.txt](https://github.com/mxe/mxe/files/1348542/gcc_i686-w64-mingw32.static.posix.txt)

**Comment by tonytheodore** [comment]
> Could it be that ${PWDCMD-pwd} evaluates to an empty string

That seems the most likely explanation, though I've never seen that construct before. Looks like some sort of `or` brace expansion:
```
$ echo ${PWDCMD-pwd}
pwd
$ export PWDCMD=foo
$ echo ${PWDCMD-pwd}
foo
$ export PWDCMD=
$ echo ${PWDCMD-pwd}

# empty line above
```

Have you got `PWDCMD` somehow set in your environment?

**Comment by abraxa** [comment]
Turns out that ${PWDCMD-pwd} is fine, it's actually the "tar xpf -" which fails for whatever reason.
As these lines are part of the install-headers-tar routine and there are alternate install-headers-cp and -cpio routines, I tried whether install-headers-cp works - it does.

I'm just submitting my workaround in case someone else runs into the same issue. Thanks for the assistance, Tony!

Workaround: https://gist.github.com/abraxa/28a307bf4ddc65f12f3db2eb8e3bf518 - put patch into mxe/src/
